x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/IdentifierBuilder.java

```diff
@@ -138,4 +147,20 @@ private static void validateIndexPattern(String indexPattern, EsqlBaseParser.Ind
     private static String removeExclusion(String indexPattern) {
         return indexPattern.charAt(0) == EXCLUSION.charAt(0) ? indexPattern.substring(1) : indexPattern;
     }
+
+    private static boolean patternExceedsMaxIndexSeparators(String pattern, int maxAllowedSeparators) {
+        int seperatorsCount = 0;
+        boolean inDateTime = false;
+        for (char ch : pattern.toCharArray()) {
+            if (ch == '<') {
+                inDateTime = true;
+            } else if (ch == '>') {
+                inDateTime = false;
+            } else if (ch == REMOTE_CLUSTER_INDEX_SEPARATOR && inDateTime == false) {
+                seperatorsCount++;
```

Comment:
There is no need to go through the entire loop and wait until the end to return `seperatorsCount > maxAllowedSeparators`. Do this comparison here, after the increment, and return early if `seperatorsCount > maxAllowedSeparators`. Otherwise, after the loop, `return false`.